libmiemu/controller.cpp

```diff
@@ -1 +1,379 @@
 #include "controller.h"
+
+void Controller::signals_in(uint16_t instruction)
+{
+	_instruction = instruction;
+	_opcode = _instruction & 0xF000;
+}
+
+Controls Controller::controls_out() const
+{
+	Controls all_controls;
+	IFControls if_controls;
+	IDControls id_controls;
+	EXControls ex_controls;
+	MEMControls mem_controls;
+	WBControls wb_controls;
+
+	switch (_opcode) {
```

Comment:
GCC warns on most of your case statements here, saying that they're out of range.  This is true because the constants are improperly defined.